Deprecation warnings when building Cesium for Unreal

I'm seeing a couple deprecation warnings when building Cesium for Unreal

> 1>C:\...\Plugins\cesium-unreal\Source\CesiumEditor\Private\CesiumIonTokenTroubleshooting.cpp(326): warning C4996: 'FEditorStyle::GetBrush': FEditorStyle::GetBrush() is deprecated, use FAppStyle::GetBrush() instead. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.

> 1>C:\...\Plugins\cesium-unreal\Source\CesiumRuntime\Private\CesiumTextureUtility.cpp(362): warning C4996: 'RHICreateTexture2D': FRHITexture2D is deprecated, please use RHICreateTexture(const FRHITextureCreateDesc&). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.

--- a/Source/CesiumEditor/Private/CesiumIonTokenTroubleshooting.cpp
+++ b/Source/CesiumEditor/Private/CesiumIonTokenTroubleshooting.cpp
@@ -18,6 +18,7 @@
 #include "Widgets/Layout/SBorder.h"
 #include "Widgets/Layout/SHeader.h"
 #include "Widgets/Text/STextBlock.h"
+#include "CesiumCommon.h"
 
 using namespace CesiumIonClient;
 

--- a/Source/CesiumEditor/Private/CesiumIonTokenTroubleshooting.cpp
+++ b/Source/CesiumEditor/Private/CesiumIonTokenTroubleshooting.cpp
@@ -347,7 +347,7 @@ void CesiumIonTokenTroubleshooting::Construct(const FArguments& InArgs) {
             .ClientSize(FVector2D(800, 600))
                 [SNew(SBorder)
                      .Visibility(EVisibility::Visible)
-                     .BorderImage(FEditorStyle::GetBrush("Menu.Background"))
+                     .BorderImage(FAppStyle::GetBrush("Menu.Background"))
                      .Padding(FMargin(10.0f, 20.0f, 10.0f, 20.0f))
                          [SNew(STextBlock)
                               .AutoWrapText(true)
@@ -510,7 +510,7 @@ void CesiumIonTokenTroubleshooting::Construct(const FArguments& InArgs) {
           .ClientSize(FVector2D(800, 600))
               [SNew(SBorder)
                    .Visibility(EVisibility::Visible)
-                   .BorderImage(FEditorStyle::GetBrush("Menu.Background"))
+                   .BorderImage(FAppStyle::GetBrush("Menu.Background"))
                    .Padding(
                        FMargin(10.0f, 10.0f, 10.0f, 10.0f))[pMainVerticalBox]]);
 }

--- a/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
+++ b/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
@@ -102,7 +102,7 @@ void UCesiumSubLevelSwitcherComponent::SetTargetSubLevel(
   }
 }
 
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_3_OR_HIGHER
 #define StreamState ELevelStreamingState
 #else
 #define StreamState ULevelStreaming::ECurrentState
@@ -140,7 +140,7 @@ void UCesiumSubLevelSwitcherComponent::TickComponent(
             this->_getLevelStreamingForSubLevel(pSubLevel);
         StreamState state =
             IsValid(pStreaming)
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_3_OR_HIGHER
                 ? pStreaming->GetLevelStreamingState() //->GetCurrentState()
 #else
                 ? pStreaming->GetCurrentState()
@@ -174,7 +174,7 @@ void UCesiumSubLevelSwitcherComponent::TickComponent(
     this->_doExtraChecksOnNextTick = false;
   }
 
-#ifdef WITH_EDITOR
+#if WITH_EDITOR
   UWorld* pWorld = this->GetWorld();
   if (!IsValid(pWorld))
     return;
@@ -211,7 +211,7 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
 
     StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_3_OR_HIGHER
       state = pStreaming->GetLevelStreamingState();
       #else
       state = pStreaming->GetCurrentState();
@@ -278,7 +278,7 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
 
     StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER 
+#if ENGINE_VERSION_5_3_OR_HIGHER 
       state = pStreaming->GetLevelStreamingState();
       #else
 

--- a/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
+++ b/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
@@ -7,6 +7,7 @@
 #include "LevelInstance/LevelInstanceLevelStreaming.h"
 #include "LevelInstance/LevelInstanceSubsystem.h"
 #include "Runtime/Launch/Resources/Version.h"
+#include "CesiumCommon.h"
 
 #if WITH_EDITOR
 #include "Editor.h"

--- a/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
+++ b/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
@@ -102,6 +102,12 @@ void UCesiumSubLevelSwitcherComponent::SetTargetSubLevel(
   }
 }
 
+#if ENGINE_VERSION_5_3_OR_HIGHER
+#define StreamState ELevelStreamingState
+#else
+#define StreamState ULevelStreaming::ECurrentState
+#endif
+
 void UCesiumSubLevelSwitcherComponent::TickComponent(
     float DeltaTime,
     enum ELevelTick TickType,
@@ -129,25 +135,32 @@ void UCesiumSubLevelSwitcherComponent::TickComponent(
         if (pSubLevel == this->_pCurrent || pSubLevel == this->_pTarget)
           continue;
 
+
         ULevelStreaming* pStreaming =
             this->_getLevelStreamingForSubLevel(pSubLevel);
-        ULevelStreaming::ECurrentState state =
-            IsValid(pStreaming) ? pStreaming->GetCurrentState()
-                                : ULevelStreaming::ECurrentState::Unloaded;
+        StreamState state =
+            IsValid(pStreaming)
+#if ENGINE_VERSION_5_3_OR_HIGHER
+                ? pStreaming->GetLevelStreamingState() //->GetCurrentState()
+#else
+                ? pStreaming->GetCurrentState()
+#endif
+                : StreamState::Unloaded;
+
         switch (state) {
-        case ULevelStreaming::ECurrentState::Loading:
-        case ULevelStreaming::ECurrentState::MakingInvisible:
-        case ULevelStreaming::ECurrentState::MakingVisible:
+        case StreamState::Loading:
+        case StreamState::MakingInvisible:
+        case StreamState::MakingVisible:
           anyLevelsStillLoaded = true;
           break;
-        case ULevelStreaming::ECurrentState::FailedToLoad:
-        case ULevelStreaming::ECurrentState::LoadedNotVisible:
-        case ULevelStreaming::ECurrentState::LoadedVisible:
+        case StreamState::FailedToLoad:
+        case StreamState::LoadedNotVisible:
+        case StreamState::LoadedVisible:
           pSubLevel->UnloadLevelInstance();
           anyLevelsStillLoaded = true;
           break;
-        case ULevelStreaming::ECurrentState::Removed:
-        case ULevelStreaming::ECurrentState::Unloaded:
+        case StreamState::Removed:
+        case StreamState::Unloaded:
           break;
         }
       }
@@ -196,10 +209,14 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
     ULevelStreaming* pStreaming =
         this->_getLevelStreamingForSubLevel(this->_pCurrent.Get());
 
-    ULevelStreaming::ECurrentState state =
-        ULevelStreaming::ECurrentState::Unloaded;
+    StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
+#if ENGINE_VERSION_5_3_OR_HIGHER
+      state = pStreaming->GetLevelStreamingState();
+      #else
       state = pStreaming->GetCurrentState();
+      #endif
+      
     } else if (this->_pCurrent->GetWorldAsset().IsNull()) {
       // There is no level associated with the target at all, so mark it
       // unloaded but also deactivate it for the benefit of the Editor UI.
@@ -207,9 +224,9 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
     }
 
     switch (state) {
-    case ULevelStreaming::ECurrentState::Loading:
-    case ULevelStreaming::ECurrentState::MakingInvisible:
-    case ULevelStreaming::ECurrentState::MakingVisible:
+    case StreamState::Loading:
+    case StreamState::MakingInvisible:
+    case StreamState::MakingVisible:
       // Wait for these transitions to finish before doing anything further.
       // TODO: maybe we can cancel these transitions somehow?
       UE_LOG(
@@ -220,9 +237,9 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
           *GetActorLabel(this->_pCurrent.Get()));
       this->_isTransitioningSubLevels = true;
       break;
-    case ULevelStreaming::ECurrentState::FailedToLoad:
-    case ULevelStreaming::ECurrentState::LoadedNotVisible:
-    case ULevelStreaming::ECurrentState::LoadedVisible:
+    case StreamState::FailedToLoad:
+    case StreamState::LoadedNotVisible:
+    case StreamState::LoadedVisible:
       UE_LOG(
           LogCesium,
           Display,
@@ -231,8 +248,8 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
       this->_isTransitioningSubLevels = true;
       this->_pCurrent->UnloadLevelInstance();
       break;
-    case ULevelStreaming::ECurrentState::Removed:
-    case ULevelStreaming::ECurrentState::Unloaded:
+    case StreamState::Removed:
+    case StreamState::Unloaded:
       UE_LOG(
           LogCesium,
           Display,
@@ -259,20 +276,24 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
     ULevelStreaming* pStreaming =
         this->_getLevelStreamingForSubLevel(this->_pTarget.Get());
 
-    ULevelStreaming::ECurrentState state =
-        ULevelStreaming::ECurrentState::Unloaded;
+    StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
+#if ENGINE_VERSION_5_3_OR_HIGHER 
+      state = pStreaming->GetLevelStreamingState();
+      #else
+
       state = pStreaming->GetCurrentState();
+      #endif
     } else if (this->_pTarget.Get()->GetWorldAsset().IsNull()) {
       // There is no level associated with the target at all, so mark it failed
       // to load because this is as loaded as it will ever be.
-      state = ULevelStreaming::ECurrentState::FailedToLoad;
+      state = StreamState::FailedToLoad;
     }
 
     switch (state) {
-    case ULevelStreaming::ECurrentState::Loading:
-    case ULevelStreaming::ECurrentState::MakingInvisible:
-    case ULevelStreaming::ECurrentState::MakingVisible:
+    case StreamState::Loading:
+    case StreamState::MakingInvisible:
+    case StreamState::MakingVisible:
       // Wait for these transitions to finish before doing anything further.
       UE_LOG(
           LogCesium,
@@ -282,9 +303,9 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
           *GetActorLabel(this->_pTarget.Get()));
       this->_isTransitioningSubLevels = true;
       break;
-    case ULevelStreaming::ECurrentState::FailedToLoad:
-    case ULevelStreaming::ECurrentState::LoadedNotVisible:
-    case ULevelStreaming::ECurrentState::LoadedVisible:
+    case StreamState::FailedToLoad:
+    case StreamState::LoadedNotVisible:
+    case StreamState::LoadedVisible:
       // Loading complete!
       UE_LOG(
           LogCesium,
@@ -301,8 +322,8 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
         this->_isTransitioningSubLevels = true;
       }
       break;
-    case ULevelStreaming::ECurrentState::Removed:
-    case ULevelStreaming::ECurrentState::Unloaded:
+    case StreamState::Removed:
+    case StreamState::Unloaded:
       // Start loading this level
       UE_LOG(
           LogCesium,

--- a/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
+++ b/Source/CesiumRuntime/Private/CesiumSubLevelSwitcherComponent.cpp
@@ -102,7 +102,7 @@ void UCesiumSubLevelSwitcherComponent::SetTargetSubLevel(
   }
 }
 
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER
 #define StreamState ELevelStreamingState
 #else
 #define StreamState ULevelStreaming::ECurrentState
@@ -140,7 +140,7 @@ void UCesiumSubLevelSwitcherComponent::TickComponent(
             this->_getLevelStreamingForSubLevel(pSubLevel);
         StreamState state =
             IsValid(pStreaming)
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER
                 ? pStreaming->GetLevelStreamingState() //->GetCurrentState()
 #else
                 ? pStreaming->GetCurrentState()
@@ -174,7 +174,7 @@ void UCesiumSubLevelSwitcherComponent::TickComponent(
     this->_doExtraChecksOnNextTick = false;
   }
 
-#if WITH_EDITOR
+#ifdef WITH_EDITOR
   UWorld* pWorld = this->GetWorld();
   if (!IsValid(pWorld))
     return;
@@ -211,7 +211,7 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
 
     StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER
       state = pStreaming->GetLevelStreamingState();
       #else
       state = pStreaming->GetCurrentState();
@@ -278,7 +278,7 @@ void UCesiumSubLevelSwitcherComponent::_updateSubLevelStateGame() {
 
     StreamState state = StreamState::Unloaded;
     if (IsValid(pStreaming)) {
-#if ENGINE_VERSION_5_3_OR_HIGHER 
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER 
       state = pStreaming->GetLevelStreamingState();
       #else
 

--- a/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
+++ b/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
@@ -27,6 +27,7 @@
 #include <memory>
 #include <stb_image_resize.h>
 #include <variant>
+#include "CesiumCommon.h"
 
 using namespace CesiumGltf;
 

--- a/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
+++ b/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
@@ -345,7 +345,7 @@ public:
       // RHICreateTexture2D can actually copy over all the mips in one shot,
       // but it expects a particular memory layout. Might be worth configuring
       // Cesium Native's mip-map generation to obey a standard memory layout.
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_2_OR_HIGHER
       rhiTexture = RHICreateTexture(
           FRHITextureCreateDesc::Create2D(createInfo.DebugName)
               .SetExtent(int32(this->_width), int32(this->_height))
@@ -889,7 +889,12 @@ UTexture2D* loadTextureGameThreadPart(LoadedTextureResult* pHalfLoadedTexture) {
   ([pTexture, pCesiumTextureResource](FRHICommandListImmediate& RHICmdList) {
     pCesiumTextureResource->SetTextureReference(
         pTexture->TextureReference.TextureReferenceRHI);
+    #if ENGINE_VERSION_5_3_OR_HIGHER 
+    pCesiumTextureResource->InitResource(FRHICommandListImmediate::Get()); //Init Resource now requires a command list.
+    #else
     pCesiumTextureResource->InitResource();
+    #endif
+
   });
 
   return pTexture;

--- a/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
+++ b/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
@@ -273,7 +273,7 @@ public:
 
   uint32 GetSizeY() const override { return this->_height; }
 
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER
   virtual void InitRHI(FRHICommandListBase& RHICmdList) override {
 #else
   virtual void InitRHI() override {
@@ -345,7 +345,7 @@ public:
       // RHICreateTexture2D can actually copy over all the mips in one shot,
       // but it expects a particular memory layout. Might be worth configuring
       // Cesium Native's mip-map generation to obey a standard memory layout.
-#if ENGINE_VERSION_5_2_OR_HIGHER
+#ifdef ENGINE_VERSION_5_2_OR_HIGHER
       rhiTexture = RHICreateTexture(
           FRHITextureCreateDesc::Create2D(createInfo.DebugName)
               .SetExtent(int32(this->_width), int32(this->_height))
@@ -414,7 +414,7 @@ FTexture2DRHIRef createAsyncTextureAndWait(
     ETextureCreateFlags Flags,
     void** InitialMipData,
     uint32 NumInitialMips) {
-#if ENGINE_VERSION_5_3_OR_HIGHER
+#ifdef ENGINE_VERSION_5_3_OR_HIGHER
   FGraphEventRef CompletionEvent;
 
   FTexture2DRHIRef result = RHIAsyncCreateTexture2D(
@@ -889,7 +889,7 @@ UTexture2D* loadTextureGameThreadPart(LoadedTextureResult* pHalfLoadedTexture) {
   ([pTexture, pCesiumTextureResource](FRHICommandListImmediate& RHICmdList) {
     pCesiumTextureResource->SetTextureReference(
         pTexture->TextureReference.TextureReferenceRHI);
-    #if ENGINE_VERSION_5_3_OR_HIGHER 
+    #ifdef ENGINE_VERSION_5_3_OR_HIGHER 
     pCesiumTextureResource->InitResource(FRHICommandListImmediate::Get()); //Init Resource now requires a command list.
     #else
     pCesiumTextureResource->InitResource();

--- a/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
+++ b/Source/CesiumRuntime/Private/CesiumTextureUtility.cpp
@@ -273,7 +273,7 @@ public:
 
   uint32 GetSizeY() const override { return this->_height; }
 
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_3_OR_HIGHER
   virtual void InitRHI(FRHICommandListBase& RHICmdList) override {
 #else
   virtual void InitRHI() override {
@@ -345,7 +345,7 @@ public:
       // RHICreateTexture2D can actually copy over all the mips in one shot,
       // but it expects a particular memory layout. Might be worth configuring
       // Cesium Native's mip-map generation to obey a standard memory layout.
-#ifdef ENGINE_VERSION_5_2_OR_HIGHER
+#if ENGINE_VERSION_5_2_OR_HIGHER
       rhiTexture = RHICreateTexture(
           FRHITextureCreateDesc::Create2D(createInfo.DebugName)
               .SetExtent(int32(this->_width), int32(this->_height))
@@ -414,7 +414,7 @@ FTexture2DRHIRef createAsyncTextureAndWait(
     ETextureCreateFlags Flags,
     void** InitialMipData,
     uint32 NumInitialMips) {
-#ifdef ENGINE_VERSION_5_3_OR_HIGHER
+#if ENGINE_VERSION_5_3_OR_HIGHER
   FGraphEventRef CompletionEvent;
 
   FTexture2DRHIRef result = RHIAsyncCreateTexture2D(
@@ -889,7 +889,7 @@ UTexture2D* loadTextureGameThreadPart(LoadedTextureResult* pHalfLoadedTexture) {
   ([pTexture, pCesiumTextureResource](FRHICommandListImmediate& RHICmdList) {
     pCesiumTextureResource->SetTextureReference(
         pTexture->TextureReference.TextureReferenceRHI);
-    #ifdef ENGINE_VERSION_5_3_OR_HIGHER 
+    #if ENGINE_VERSION_5_3_OR_HIGHER 
     pCesiumTextureResource->InitResource(FRHICommandListImmediate::Get()); //Init Resource now requires a command list.
     #else
     pCesiumTextureResource->InitResource();

